src/SW_Control.c

```diff
@@ -518,6 +525,86 @@ void SW_CTL_run_current_year(SW_ALL* sw, SW_OUTPUT_POINTERS* SW_OutputPtrs,
   #endif
 }
 
+/**
+@brief Calls 'SW_CTL_run_current_year' over an array of simulated years
+          as specified by the given spinup scope and duration which
+          then calls 'SW_SWC_water_flow' for each day.
+
+@param[in,out] sw Comprehensive struct of type SW_ALL containing all
+  information in the simulation
+@param[in,out] SW_OutputPtrs SW_OUTPUT_POINTERS of size SW_OUTNKEYS which
+  hold pointers to subroutines for output keys
+@param[in,out] LogInfo Holds information dealing with logfile output
+*/
+void SW_CTL_run_spinup(SW_ALL* sw, SW_OUTPUT_POINTERS* SW_OutputPtrs,
+                      LOG_INFO* LogInfo) {
+  // spinup stuff
+
+  unsigned int i, quotient = 0, remainder = 0;
+  int mode = sw->Model.spinup_mode,
+      yr;
+  TimeInt
+    duration = sw->Model.spinup_duration,
+    scope = sw->Model.spinup_scope,
+    finalyr = sw->Model.startyr + scope,
+    years[ duration ];
+
+  #ifdef SWDEBUG
+  int debug = 0;
+  #endif
+  
+  switch ( mode ) {
+    case 2:
+      // initialize structured array
+      if ( duration <= scope ) {
+        // 1:m
+        yr = sw->Model.startyr;
+        for ( i = 0; i < duration; i++ ) {
+          years[ i ] = yr + i;
+        }
+      }
+      else {
+        // { {1:n}_(m//n), 1:(m%n) }
+        quotient = duration / scope;
+        remainder = duration % scope;
+        yr = sw->Model.startyr;
+        for ( i = 0; i < quotient * scope; i ++ ) {
+          years[ i ] = yr + ( i % scope );
+        }
+        for ( i = 0; i < remainder; i++) {
+          years[ i + ( scope * quotient ) ] = yr + i;
+        }
+      }
+
+      break;
+    default: // same as case 1
+      // initialize random array
+      for ( i = 0; i < duration; i++ ) {
+        yr = RandUniIntRange( sw->Model.startyr,
+                                      finalyr,

```

Comment:
We define `spinup_scope` as "first N years of simulation" ; however, we set `finalyr = sw->Model.startyr + scope` and `RandUniIntRange()` includes the bounds. This results that mode 1 uses the first N + 1 years. Thus, we need `yr = RandUniIntRange( sw->Model.startyr, finalyr - 1, ...)` -- or `finalyr = sw->Model.startyr + scope - 1`.